src/braket/circuits/noise.py

```diff
@@ -331,85 +453,152 @@ def __init__(
+            self._parameters.append(param)
+            return 0
+        else:
+            if not isinstance(param, float):
+                raise TypeError(f"{paramName} must be float type")
+            if not (param <= 1.0 and param >= 0.0):
+                raise ValueError(f"{paramName} must be a real number in the interval [0,1]")
+            self._parameters.append(float(param))
+            return param
 
     @property
-    def probX(self) -> float:
+    def probX(self) -> Union[FreeParameter, float]:
         """
         Returns:
-            probX (float): The probability of a Pauli X error.
+            probX (Union[FreeParameter, float]): The probability of a Pauli X error.
         """
-        return self._probX
+        return self._parameters[0]
 
     @property
-    def probY(self) -> float:
+    def probY(self) -> Union[FreeParameter, float]:
         """
         Returns:
-            probY (float): The probability of a Pauli Y error.
+            probY (Union[FreeParameter, float]): The probability of a Pauli Y error.
         """
-        return self._probY
+        return self._parameters[1]
 
     @property
-    def probZ(self) -> float:
+    def probZ(self) -> Union[FreeParameter, float]:
         """
         Returns:
-            probZ (float): The probability of a Pauli Z error.
+            probZ (Union[FreeParameter, float]): The probability of a Pauli Z error.
         """
-        return self._probZ
+        return self._parameters[2]
 
     def __repr__(self):
-        return f"{self.name}('probX': {self.probX}, 'probY': {self.probY}, \
-'probZ': {self.probZ}, 'qubit_count': {self.qubit_count})"
+        return f"{self.name}('probX': {self._parameters[0]}, 'probY': {self._parameters[1]}, \
+'probZ': {self._parameters[2]}, 'qubit_count': {self.qubit_count})"
+
+    def __str__(self):
+        return f"{self.name}({self._parameters[0]}, {self._parameters[1]}, {self._parameters[2]})"
+
+    def __eq__(self, other):
+        if isinstance(other, type(self)):
+            return self.name == other.name and self._parameters == other._parameters
+        return False
+
+    @property
+    def parameters(self) -> List[Union[FreeParameter, float]]:
+        """
+        Returns the free parameters associated with the object.
+
+        Returns:
+            Union[FreeParameter, float]: Returns the free parameters or fixed value
+            associated with the object.
+        """
+        return self._parameters
+
+    def to_dict(self) -> dict:
+        """
+        Converts a this object into a dictionary representation.
 
+        Returns:
+            dict: A dictionary object that represents this object. It can be converted back
+            into this object using the `from_dict()` method.
+        """
+        return {
+            "__class__": self.__class__.__name__,
+            "probX": self.probX,
+            "probY": self.probY,
+            "probZ": self.probZ,
```

Comment:
I wanted some way to be able to serialize and deserialize. I thought I would need to keep track of the type (class) of each object and the values, and recursively call all of the children. 

If I can do that with __dict__, I would be happy to. But it seems like I would just need to implement __dict__ (rather than  to_dict()) for each class instead. Would that be better?